### ISSUE TITLE
ScheduledReporter.start() starts with undesired delay

### ISSUE THREAD

**Issue by sofax** [dropwizard/metrics#998]
`ScheduledReporter.start()` incorrectly uses the interval size for the start delay, too:

```
    public void start(long period, TimeUnit unit) {
        executor.scheduleAtFixedRate(new Runnable() {
            @Override
            public void run() {
                try {
                    report();
                } catch (RuntimeException ex) {
                    LOG.error("RuntimeException thrown from {}#report. Exception was suppressed.", ScheduledReporter.this.getClass().getSimpleName(), ex);
                }
            }
        }, period, period, unit);
    }
```

This is a serious problem when the interval is sufficiently big. I suggest to add an alternative version of `ScheduledReporter.start()` to fix the bug without affecting code that relies on this behavior:

```
    public void start(long period, long initialDelay, TimeUnit unit) {
        executor.scheduleAtFixedRate(new Runnable() {
            @Override
            public void run() {
                try {
                    report();
                } catch (RuntimeException ex) {
                    LOG.error("RuntimeException thrown from {}#report. Exception was suppressed.", ScheduledReporter.this.getClass().getSimpleName(), ex);
                }
            }
        }, initialDelay, period, unit);
    }
```

**Comment by sofax** [comment]
I added the suggested fix to the fork 
https://github.com/sofax/metrics

**Comment by sofax** [comment]
This is the pull request for it:
https://github.com/dropwizard/metrics/pull/999

**Comment by ryantenney** [comment]
Will track in #999

**Comment by arteam** [comment]
Closed via #999
